### DISCUSSION TITLE
Bluk Insert

### DISCUSSION THREAD

**Discussion by go4cas** [pocketbase/pocketbase#4804] [Q&A]
When extending using Javascript, I have an array of objects (the keys of the objects replicate the column names of a collection). Is there a shorthand way to bulk insert the entire array? Or, is the only way to loop over the array, and then `new Record()` and `$app.dao().saveRecord()` for each object in the array?

**Comment by ganigeorgiev** [comment]
You need to loop over the array items and call `saveRecord` for each but note that you need to use the `txDao`. For an example you could check https://pocketbase.io/docs/js-records/#transaction.

**Reply by go4cas** [reply]
Thanks, @ganigeorgiev! I'll try the `txDao`. I did see a `load` method ("_bulk set fields from a map_"), but the documentation is not clear on exactly how to use it, with examples. What this is use case of this method?

**Reply by ganigeorgiev** [reply]
The load method is to load/populate multiple fields in a single record at once, eg.:
```js
record.load({
    "title":       "...",
    "description": "..."
})
// is the the same as
record.set("title", "...")
record.set("description", "...")
```
